### PR TITLE
Add referred_by to config

### LINKED ISSUE #1500: Add additional referred by fields to config
Thanks for creating an issue! Please fill out this form so we can be sure to have all the information we need, and to minimize back and forth.

* **What are we trying to do?**

While we have a pretty solid list of `referred_by` fields, we don't cover every case, and BRAF is interested in reshaping this list. So we should make it fund-configurable.

* **What feature or behavior is this required for?**

Better tracking of who a fund is working with and where their patients are coming from.

* **How could we solve this issue? (Not knowing is okay!)**

* Add it as an option in `app/models/config.rb`
* Have that select dropdown in `patient_information` use the stock configs + whatever else they're adding

This is pretty similar to what we're doing with Language, IIRC, so you can probably adopt a lot of it from that.

* **Anything else?**

Stems from #1465

### PR DESCRIPTION
**Pull request by tingaloo**
I rule and have completed some work on Case Manager that's ready for review!

Vacate the code.

This pull request makes the following changes:
* Add referred_by to config

It relates to the following issue #s: 
* Fixes #1500

### CODE REVIEW

File: app/helpers/patients_helper.rb [hunk 1]
```diff
@@ -19,6 +19,11 @@ def language_options
     standard_options + Config.find_or_create_by(config_key: 'language').options
   end
 
+  def referred_by_options
+    standard_options = [[nil]]
```
Comment (colinxfleming) [inline app/helpers/patients_helper.rb]: Let's port over the existing list as our standard options, and let funds staple on an additional set if they want via the config.

```suggestion
    standard_options = [nil, 'Clinic', 'Crime victim advocacy center',
     "#{FUND} website or social media",
     'Domestic violence crisis/intervention org', 'Family member', 'Friend',
     'Google/Web search', 'Homeless shelter', 'Legal clinic', 'NAF', 'NNAF',
     'Other abortion fund', 'Previous patient', 'School',
     'Sexual assault crisis org', 'Youth outreach']
```

**Comment by vraravam** [comment]
Shouldn't this also have a UI change where the user can set this in the admin section?

**Comment by colinxfleming** [comment]
I spruced this up some. @lomky or @DarthHater a review at yr leisure pls!

**Comment by colinxfleming** [comment]
The UI thing is an A++ flag. rep'd that in #1559

**Pull request merged**